### DISCUSSION TITLE
Composition

### DISCUSSION THREAD

**Discussion by ahmed-adly-khalil** [gzuidhof/tygo#37] [Ideas]
the following structure would be serialized to JSON as a flat object if we use the standard library, 

```
type Animal struct {
    Sound string
}

type Dog struct {
    Animal
    Breed string
}
```

however in tygo we have to add `json:"animal"` to the Dog type in order to serialize it and this would result in a field representation, is there a way to tell tygo to merge the composed struct fields with the main struct?

**Comment by moeshin** [comment]
```go
type Animal struct {
	Sound string
}

type Dog struct {
	Animal `tstype:",extends"`
	Breed  string
}

```

```ts
export interface Animal {
  Sound: string;
}
export interface Dog extends Animal {
  Breed: string;
}
```

docs: https://github.com/gzuidhof/tygo/tree/342962e8fe07318cb925af45c3f804e8c9fcb930#inheritance

**Reply by ahmed-adly-khalil** [reply]
thanks
